File: dataset/human/java/file_1298.java
import java.util.*;

public class Main {

    static final int MOD = 1000000007;

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int a[] = new int[n];
        long sum[] = new long[n];
        for(int i=0;i<n;i++)a[i]=sc.nextInt();
        sum[0]=a[0];
        for(int i=1;i<n;i++)sum[i]=sum[i-1]+a[i];
        long ans = Long.MAX_VALUE;
        for(int i=0;i<n-1;i++){
            long x= sum[i];
            long y = sum[n-1]-sum[i];
            //System.out.println(x+" "+y);
            ans = Math.min(ans,Math.abs(x-y));
        }
        System.out.println(ans);
    }
}

File: dataset/human/python/file_1298.py
import sys
import numpy as np

readline = sys.stdin.readline

def main():
    N = int(input())
    A = list(map(int, readline().split()))

    base0 = A[0]
    base1 = sum(A) - base0

    ans = abs(base1 - base0)

    for i in range(1,N-1):
        base0 += A[i]
        base1 -= A[i]
        ans = min(abs(base1 - base0), ans)

    print(ans)


if __name__ == "__main__":
    main()
